### PR TITLE
chore: add err code for unknown signal server on dial

### PR DESCRIPTION
**Pull request by vasco-santos**
Per https://github.com/ipfs/js-ipfs/pull/3633

### CODE REVIEW

**Review by achingbrain — APPROVED** [review]
LGTM.

The test may be better in the form of 

```js
return expect(ws1.dial(maOffline)).to.eventually.be.rejected.and.have.property('code', 'ERR_UNKNOWN_SIGNAL_SERVER')
```

since then any future broken behaviour would fail on an assertion instead of needing to throw an error but it's a minor nit.

File: test/transport/dial.js [hunk 1]
```diff
@@ -88,15 +88,14 @@ module.exports = (create) => {
...
-        await ws1.dial(maOffline)
-      } catch (err) {
-        expect(err).to.be.an.instanceOf(Error)
-        expect(err).to.have.property('code', 'ERR_SIGNALLING_FAILED')
-        return
-      }
+      return expect(ws1.dial(maOffline)).to.eventually.be.rejected.and.have.property('code', 'ERR_SIGNALLING_FAILED')
+    })
+
+    it('dial unknown signal server, check promise rejected', async function () {
```
Comment (achingbrain) [inline test/transport/dial.js:94]: I'm surprised this didn't cause linting to fail - there's no await in this async function..

Comment (achingbrain) [reply test/transport/dial.js:94]: Nor in the `'dial offline / non-exist()ent node on IPv4, check promise rejected'` test

Comment (vasco-santos) [reply test/transport/dial.js:94]: interesting 🤔 That is true

**Pull request merged**